src/post-types.php

```diff
@@ -153,22 +152,37 @@ public function handle_submit() {
 			$this->set_bool_option( 'enable_gutenberg_videos' );
 		}
 
+		// If we've now enabled the post types, make sure they work.
+		if ( $this->option->get( 'enable_post_types' ) && ! post_type_exists( 'book' ) ) {
+			$this->register_post_types();
+
+			// Hook this to shutdown so we're certain all the required post types have been registered.
+			add_action( 'shutdown', [ $this, 'flush_rewrite_rules' ] );
+		}
+
 		wp_safe_redirect(
 			self_admin_url(
 				'tools.php?page=' .
-				apply_filters( 'yoast_version_control_admin_page', '' )
+				apply_filters( 'Yoast\WP\Test_Helper\admin_page', '' )
 			)
 		);
 	}
 
+	/**
+	 * Flushes the rewrite rules on the required action.
+	 */
+	public function flush_rewrite_rules() {
+		\flush_rewrite_rules();
+	}
+
 	/**
 	 * Sets a boolean option based on a POST parameter.
 	 *
 	 * @param string $option The option to check and set.
 	 */
 	private function set_bool_option( $option ) {
 		// The nonce is checked in the handle_submit function.
-		// phpcs:ignore WordPress.CSRF.NonceVerification.NoNonceVerification
+		// @phpcs:ignore WordPress.Security.NonceVerification.Missing
```

Comment:
The `@` is not needed.
```suggestion
		// phpcs:ignore WordPress.Security.NonceVerification.Missing
```